default.nix

```diff
@@ -6,72 +6,109 @@ with pkgs;
 
 let
   root = ./.;
-  mainnet = {
+  protocol005 = rec {
+    protocolName = "005_PsBabyM1";
+    binarySuffix = builtins.replaceStrings [ "_" ] [ "-" ] protocolName;
+  };
+  mainnet = rec {
     rev = "94f779a7";
     sha256 = "16lxilng5q8fr2ll6h4hf7wlvac6nmw4cx10cbgzj5ks090bl97r";
     patchFile = ./nix/fix-mainnet.patch;
+    protocol = protocol005;
   };
-  babylonnet = {
+  babylonnet = rec {
     rev = "b8731913";
     sha256 = "1pakf1s6bg76fq42mb8fj1immz9g9wwimd522cpx8k28zf0hkl5i";
     patchFile = ./nix/fix-babylonnet.patch;
+    protocol = protocol005;
   };
-  tezos-client-static-mainnet = import ./nix/static.nix mainnet;
-  tezos-client-static-babylonnet = import ./nix/static.nix babylonnet;
-  binary-mainnet = "${tezos-client-static-mainnet}/bin/tezos-client";
-  binary-babylonnet = "${tezos-client-static-babylonnet}/bin/tezos-client";
-  # Hopefully, there will always be a single LICENSE for all binaries and branches
-  licenseFile = "${tezos-client-static-mainnet}/LICENSE";
-  packageDesc-mainnet = {
-    inherit licenseFile;
-    project = "tezos-client-mainnet";
-    version = toString timestamp;
-    bin = binary-mainnet;
-    arch = "amd64";
-    license = "MIT";
-    dependencies = "";
-    maintainer = "Serokell https://serokell.io";
-    description = "CLI client for interacting with tezos blockchain";
-    gitRevision = mainnet.rev;
-    branchName = "mainnet";
-  };
-
-  packageDesc-babylonnet = packageDesc-mainnet // {
-    project = "tezos-client-babylonnet";
-    bin = binary-babylonnet;
-    gitRevision = babylonnet.rev;
-    branchName = "babylonnet";
-  };
-
-  buildDeb = import ./packageDeb.nix { inherit stdenv writeTextFile dpkg; };
-  buildRpm = packageDesc:
-  import ./packageRpm.nix { inherit stdenv writeTextFile gnutar rpm buildFHSUserEnv; }
-    (packageDesc // { arch = "x86_64"; });
+  static-nix = import ./nix/static.nix;
+  tezos-static-mainnet = static-nix mainnet;
+  tezos-static-babylonnet = static-nix babylonnet;
 
-  mainnet-rpm-package = buildRpm packageDesc-mainnet;
+  packDirectory = archiveName: pathToPack:
+    stdenv.mkDerivation rec {
+      name = "${archiveName}.tar.gz";
+      phases = "archivePhase";
+      nativeBuildInputs = [ gnutar ];
+      archivePhase = ''
+        mkdir -p ${archiveName}
```

Comment:
just figure out the right tar flags to do this without moving stuff around

Reply:
Updated, btw, now no moving or copying the stuff around